build/build.yml

```diff
@@ -0,0 +1,116 @@
+pool:
+  name: Hosted macOS
+  demands:
+  - npm
+  - node.js
+
+steps:
+- task: ArchiveFiles@1
+  displayName: 'Archive source '
+  inputs:
+    rootFolder: '$(Build.SourcesDirectory)'
+    includeRootFolder: false
+    archiveType: tar
+    archiveFile: '$(Build.BinariesDirectory)/source.tar.gz'
+
+- task: PublishBuildArtifacts@1
+  displayName: 'Publish Artifact: source'
+  inputs:
+    PathtoPublish: '$(Build.BinariesDirectory)/source.tar.gz'
+    ArtifactName: source
+
+- task: Npm@1
+  displayName: 'Use npm 6.5.0'
+  inputs:
+    command: custom
+    verbose: false
+    customCommand: 'install -g npm@6.5.0'
+
+- task: Npm@1
+  displayName: 'npm install'
+  inputs:
+    workingDir: '$(Build.SourcesDirectory)'
+    verbose: false
+
+- task: Npm@1
+  displayName: 'npm custom'
+  inputs:
+    command: custom
+    verbose: false
+    customCommand: 'install -g gulp-cli'
+
+- task: Gulp@0
+  displayName: 'gulp install'
+  inputs:
+    targets: install
+
+- bash: |
+   export BUILDMACHINE=true
+   export VsMsSqlEnv=dev
+   node node_modules/gulp/bin/gulp.js build --gulpfile $(Build.SourcesDirectory)/gulpfile.js
+  displayName: 'gulp build (dev)'
+  condition: and(succeeded(), eq(variables['release'], false))
+
+- bash: |
+   export BUILDMACHINE=true
+   export VsMsSqlEnv=production
+   node node_modules/gulp/bin/gulp.js build --gulpfile $(Build.SourcesDirectory)/gulpfile.js
+  displayName: 'gulp build (release)'
+  condition: and(succeeded(), eq(variables['release'], true))
+
+- bash: 'npm install -g vsce'
+  displayName: 'npm install vsce'
+
+- task: Gulp@0
+  displayName: 'gulp package:offline'
+  inputs:
+    targets: 'package:offline'
+
+- task: Gulp@0
+  displayName: 'gulp package:online'
+  inputs:
+    targets: 'package:online'
+
+- task: CopyFiles@2
+  displayName: 'Copy Files to: $(Build.ArtifactStagingDirectory)'
+  inputs:
+    SourceFolder: '$(Build.SourcesDirectory)'
+    Contents: '*.vsix'
+    TargetFolder: '$(Build.ArtifactStagingDirectory)'
+
+- task: PublishBuildArtifacts@1
+  displayName: 'Publish Artifact: drop'
+
+- script: |
+   brew update
```

Comment:
We should move to vscode's tester so we don't have to do this, but i guess thats a separate issue.